homeassistant/components/motionblinds_ble/config_flow.py

```diff
@@ -0,0 +1,220 @@
+        self, discovery_info: BluetoothServiceInfoBleak
+    ) -> ConfigFlowResult:
+        """Handle the bluetooth discovery step."""
+        _LOGGER.debug(
+            "Discovered Motionblinds bluetooth device: %s", discovery_info.as_dict()
+        )
+        await self.async_set_unique_id(discovery_info.address)
+        self._abort_if_unique_id_configured()
+
+        self._discovery_info = discovery_info
+        self._mac_code = get_mac_from_local_name(discovery_info.name)
+        self._display_name = DISPLAY_NAME.format(mac_code=self._mac_code)
+        self.context["local_name"] = discovery_info.name
+        self.context["title_placeholders"] = {"name": self._display_name}
+
+        return await self.async_step_confirm()
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            mac_code = user_input[CONF_MAC_CODE]
+            # Discover with BLE
+            try:
+                await self.async_discover_motionblind(mac_code)
+            except NoBluetoothAdapter:
+                return self.async_abort(reason=EXCEPTION_MAP[NoBluetoothAdapter])
+            except NoDevicesFound:
+                return self.async_abort(reason=EXCEPTION_MAP[NoDevicesFound])
+            except tuple(EXCEPTION_MAP.keys()) as e:
+                errors = {
+                    "base": (
+                        EXCEPTION_MAP[type(e)]
+                        if type(e) in EXCEPTION_MAP
+                        else str(type(e))
+                    )
+                }
+                return self.async_show_form(
+                    step_id="user", data_schema=CONFIG_SCHEMA, errors=errors
+                )
+            return await self.async_step_confirm()
+
+        scanner_count = bluetooth.async_scanner_count(self.hass, connectable=True)
+        if not scanner_count:
+            _LOGGER.error("No bluetooth adapter found")
+            return self.async_abort(reason=EXCEPTION_MAP[NoBluetoothAdapter])
+
+        return self.async_show_form(
+            step_id="user", data_schema=CONFIG_SCHEMA, errors=errors
+        )
+
+    async def async_step_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Confirm a single device."""
+        if user_input is not None:
+            self._blind_type = user_input[CONF_BLIND_TYPE]
+
+            if TYPE_CHECKING:
+                assert self._discovery_info is not None
+
+            return self.async_create_entry(
+                title=str(self._display_name),
+                data={
+                    CONF_ADDRESS: self._discovery_info.address,
+                    CONF_LOCAL_NAME: self._discovery_info.name,
+                    CONF_MAC_CODE: self._mac_code,
+                    CONF_BLIND_TYPE: self._blind_type,
```

Comment:
Can we discover the blind type without having to ask them. I'm not sure the user will know.

Reply:
Unfortunately we can't, the Motionblinds BLE app also asks the user for their blind type